### PR TITLE
completed models and associations

### PR DESCRIPTION
**Pull request by saba-mo**
### Assignee Tasks

* [ ] updated user model to include first and last name
* [ ] added order and cart models
* [ ] set up associations 

### Guidelines

Should be able to run seed!

---

_Your PR Notes Here_

### CODE REVIEW

File: server/db/models/cart.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const Sequelize = require('sequelize')
+const db = require('../db')
+
+module.exports = db.define('cart', {
+  quantity: {
+    type: Sequelize.INTEGER,
+    // look into whether this the place to add max: item inventory
+    validate: {
```
Comment (ahillary) [inline server/db/models/cart.js]: Yes

Comment (ahillary) [reply server/db/models/cart.js]: Remove comment

File: server/db/models/index.js [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
 const Product = require('./product')
+const Order = require('./order')
+const Cart = require('./cart')
 
 /**
  * If we had any associations to make, this would be a great place to put them!
  * ex. if we had another model called BlogPost, we might say:
  *
  *    BlogPost.belongsTo(User)
  */
```
Comment (ahillary) [inline server/db/models/index.js]: remove comment

File: server/db/models/index.js [hunk 3]
```diff
@@ -1,12 +1,19 @@
...
+Product.belongsToMany(Order, {through: Cart})
+Order.belongsToMany(Product, {through: Cart})
+
+User.hasOne(Order)
+Order.belongsTo(User)
 
 /**
  * We'll export all of our models here, so that any time a module needs a model,
```
Comment (ahillary) [inline server/db/models/index.js]: remove comment

File: server/db/models/user.js [hunk 4]
```diff
@@ -3,6 +3,14 @@ const Sequelize = require('sequelize')
...
 
 const User = db.define('user', {
+  firstName: {
+    type: Sequelize.STRING
+    // allowNull: false,
+  },
+  lastName: {
+    type: Sequelize.STRING
+    // allowNull: false,
+  },
```
Comment (ahillary) [inline server/db/models/user.js]: uncomment

**Review by ahillary — CHANGES_REQUESTED** [review]
see comments

**Review by abstauff — APPROVED** [review]

**Review by abstauff — APPROVED** [review]

**Review by ahillary — APPROVED** [review]

**Pull request merged**